.travis.yml

```diff
@@ -3,17 +3,28 @@ language: python
 python:
 - 2.7
 
+addons:
+  firefox: latest
+  sauce_connect: true
+  apt:
+    sources:
+    - google-chrome
+
 branches:
   only:
   - master
   - develop
 
 env:
-- RUN_BACKEND_TESTS=false RUN_FRONTEND_TESTS=false RUN_LINT=true REPORT_BACKEND_COVERAGE=false
-- RUN_BACKEND_TESTS=false RUN_FRONTEND_TESTS=true RUN_LINT=false REPORT_BACKEND_COVERAGE=false
-- RUN_BACKEND_TESTS=true RUN_FRONTEND_TESTS=false RUN_LINT=false REPORT_BACKEND_COVERAGE=false
-- RUN_BACKEND_TESTS=true RUN_FRONTEND_TESTS=false RUN_LINT=false REPORT_BACKEND_COVERAGE=true
-
+  matrix:
+  - RUN_E2E_TESTS=true
+  - RUN_BACKEND_TESTS=false RUN_FRONTEND_TESTS=false RUN_LINT=true REPORT_BACKEND_COVERAGE=false
+  - RUN_BACKEND_TESTS=false RUN_FRONTEND_TESTS=true RUN_LINT=false REPORT_BACKEND_COVERAGE=false
+  - RUN_BACKEND_TESTS=true RUN_FRONTEND_TESTS=false RUN_LINT=false REPORT_BACKEND_COVERAGE=false
+  - RUN_BACKEND_TESTS=true RUN_FRONTEND_TESTS=false RUN_LINT=false REPORT_BACKEND_COVERAGE=true
+  global:
+  - secure: CdqriDTcKaS6LMGsL+zqLGyhKv4xWzycKgBFlhVEBwgG70kOH7I+HKj29UzA6Ym1Mb2hofKVSlRkhepzuG8CgH9Rd07lUeWnERphIdaZwGYNMulVZhZM0/6Quo2hNDDDHzyhOWlM2XkmzUsaBp8M0J/tIa5A9HkSpHZ4Fj3tEAsPzqj1wewxS7iQHlOhCXZb8WzpG/WEiJH8tonPbJCZBr+ZW8C3F8DdnTrl4Boino0PBLr01rgu5sURYbPKpi1Qa3iW9DMKAA9qEbz89ofnSvt9tXz7FdNqAbJ0B0KYpmS1rU82+Lz58dMXmHFnRxAZiIBeG+Fhzl97b0fG+d/S2+KiRostXcWVbVHa/h4wyfoBtVZD0XFgSUxCKJ10T+eFJhLrBstoCfIzJ10sxHLr/hrmPpJ/Ac+v2F2/ib3pFIvHrJOmbve831HzMVTlAVVqX2C+kkr+WH1QhdUw1AMm6aYTOsE4ZxBGV6IqWOkPOn7kA/CWnqaea2xVJ7O7drkXbO5cQCVpP3X+RxF1w2tUfseJftGx8A9Ca0JRjWpNvFX2AVyuK6gEgB7fWfCmfVmLdaVtSPXNMerTWaD/jLKPwcla92+sUh9iCVEiVfrYxx63U+27YJ8hwfHfkIXuuf6iGzh5vmp04KA93KSmQmMRddN4MhuuV7t/avVXRxKxuwQ=
```

Comment:
Why two of these? (Just wondering.)

Might be worth adding some explanatory comments.